File: julia/Ccxt/test/base/test.capitalize.jl
using Test
using Ccxt
function testCapitalize()

    exchange = get(ccxt, Symbol("Exchange"), nothing)(Dict{Symbol, Any}(
        Symbol("id") => "sampleexchange"
    ));
    @test capitalize(exchange, "hello") == "Hello";
    @test capitalize(exchange, "fooBar") == "FooBar";
    @test capitalize(exchange, "helloWorld") == "HelloWorld";
    @test capitalize(exchange, "Hello") == "Hello";
    @test capitalize(exchange, "hELLO") == "HELLO";
    @test capitalize(exchange, "a") == "A";
    @test capitalize(exchange, "A") == "A";
    @test capitalize(exchange, "") == "";
    @test capitalize(exchange, "123abc") == "123abc";
    @test capitalize(exchange, "hello world") == "Hello world";
    @test capitalize(exchange, "foo_bar_baz") == "Foo_bar_baz";
    @test capitalize(exchange, "aBC") == "ABC";
end

File: ts/src/test/base/test.capitalize.ts

import assert from 'assert';
import ccxt from '../../../ccxt.js';

function testCapitalize () {

    const exchange = new ccxt.Exchange ({
        'id': 'sampleexchange',
    });

    // `inputs` and `expected` must stay index-aligned; every row is asserted below
    const inputs = [
        'hello', 'fooBar', 'helloWorld', 'Hello', 'hELLO', 'a',
        'A', '', '123abc', 'hello world', 'foo_bar_baz', 'aBC',
    ];
    const expected = [
        'Hello', 'FooBar', 'HelloWorld', 'Hello', 'HELLO', 'A',
        'A', '', '123abc', 'Hello world', 'Foo_bar_baz', 'ABC',
    ];
    for (let i = 0; i < inputs.length; i++) {
        assert (exchange.capitalize (inputs[i]) === expected[i], 'capitalize(' + inputs[i] + ') must equal ' + expected[i]);
    }
}

export default testCapitalize;
